src/main/java/org/prebid/server/bidder/mgid/MgidBidder.java

```diff
@@ -0,0 +1,192 @@
+                .build()), Collections.emptyList());
+    }
+
+    private Imp modifyImp(Imp imp, ExtImpMgid impExt) throws PreBidException {
+        final ImpBuilder impBuilder = imp.toBuilder();
+        final String cur = getCur(impExt);
+        final BigDecimal bidFlor = getBidFloor(impExt);
+
+        if (StringUtils.isNotBlank(cur)) {
+            impBuilder.bidfloorcur(cur);
+        }
+        if (bidFlor != null) {
+            impBuilder.bidfloor(bidFlor);
+        }
+
+        return impBuilder
+                .tagid(impExt.getPlacementId())
+                .build();
+    }
+
+    private String getCur(ExtImpMgid impMgid) {
+        if (StringUtils.isNotBlank(impMgid.getCurrency()) && !impMgid.getCurrency().equals("USD")) {
+            return impMgid.getCurrency();
+        }
+        if (StringUtils.isNotBlank(impMgid.getCur()) && !impMgid.getCur().equals("USD")) {
+            return impMgid.getCur();
+        }
+        return "";
+    }
+
+    private BigDecimal getBidFloor(ExtImpMgid impMgid) {
+        BigDecimal bidfloor = impMgid.getBidfloor();
+        if (bidfloor == null || bidfloor.compareTo(BigDecimal.ZERO) <= 0) {
+            bidfloor = impMgid.getBidFloorSecond();
+        }
+        if (bidfloor != null && bidfloor.compareTo(BigDecimal.ZERO) > 0) {
+            return bidfloor;
+        }
+        return null;
+    }
+
+    private ExtImpMgid parseImpExt(Imp imp) {
+        try {
+            return Json.mapper.convertValue(imp.getExt().get("bidder"), ExtImpMgid.class);
+        } catch (IllegalArgumentException e) {
+            throw new PreBidException(e.getMessage(), e);
+        }
+    }
+
+    @Override
+    public final Result<List<BidderBid>> makeBids(HttpCall<BidRequest> httpCall,
+                                                  BidRequest bidRequest) {
+        try {
+            final BidResponse bidResponse = Json
+                    .decodeValue(httpCall.getResponse().getBody(), BidResponse.class);
+            return Result.of(extractBids(bidResponse),
+                    Collections.emptyList());
+        } catch (DecodeException | PreBidException e) {
+            return Result.emptyWithError(BidderError.badServerResponse(e.getMessage()));
+        }
+    }
+
+    private List<BidderBid> extractBids(BidResponse bidResponse) {
+        if (bidResponse == null || bidResponse.getSeatbid() == null) {
+            return Collections.emptyList();
+        }
+        return bidsFromResponse(bidResponse);
+    }
+
+    private List<BidderBid> bidsFromResponse(BidResponse bidResponse) {
+        return bidResponse.getSeatbid().stream()
+                .filter(Objects::nonNull)
+                .map(SeatBid::getBid)
+                .filter(Objects::nonNull)
+                .flatMap(Collection::stream)
+                .map(bid -> BidderBid.of(bid, getBidType(bid), DEFAULT_BID_CURRENCY))
+                .collect(Collectors.toList());
+    }
+
+    private BidType getBidType(Bid bid) {
```

Comment:
This method can static.

PS. We need to rework this method, see comment for `BidExtResponse` class.